Core/NoOpRunner.Core/Shapes/Platform.cs

```diff
@@ -1,61 +1,98 @@
 ﻿using NoOpRunner.Core.Entities;
 using NoOpRunner.Core.Enums;
 using System;
+using System.Collections.Generic;
 
 namespace NoOpRunner.Core.Shapes
 {
     public class Platform : BaseShape
     {
-        private Random random = new Random();
         public int bottomPosY { get; set; } // Bottom of x platform
         public int topPosY { get; set; } // Max height of x platform
 
+        private Random randomNumber;
```

Comment:
This should be a protected variable in the base class, not injected through constructor, or a singleton, to make sure the same instance is used for every object.